src/core/effects/qgsgloweffect.cpp

```diff
@@ -150,11 +158,18 @@ void QgsGlowEffect::readProperties( const QgsStringMap &props )
   }
   mEnabled = props.value( QStringLiteral( "enabled" ), QStringLiteral( "1" ) ).toInt();
   mDrawMode = static_cast< QgsPaintEffect::DrawMode >( props.value( QStringLiteral( "draw_mode" ), QStringLiteral( "2" ) ).toInt() );
-  int level = props.value( QStringLiteral( "blur_level" ) ).toInt( &ok );
+  double level = props.value( QStringLiteral( "blur_level" ) ).toDouble( &ok );
   if ( ok )
   {
     mBlurLevel = level;
   }
+  if ( !props.contains( QStringLiteral( "blur_unit" ) ) && props.contains( QStringLiteral( "blur_level" ) ) )
+  {
+    // deal with pre blur unit era by assuming 96 dpi and converting pixel values as millimeters
+    mBlurLevel *= 0.2645;
```

Comment:
If the above conversion of blur_level to double fails, then mBlurLevel will be the default (in mm), and you'll double-convert it here. Not sure that situation can arise in reality, but this branch should probably sit inside the if (ok ) block above.